marshmallow/schema.py

```diff
@@ -838,9 +838,9 @@ def _invoke_validators(self, pass_many, data, original_data, many, field_errors=
             if many and not pass_many:
                 for idx, item in enumerate(data):
                     try:
-                        self._unmarshal.run_validator(validator,
-                                                  item, original_data, self.fields, many=many,
-                                                  index=idx, pass_original=pass_original)
+                        self._unmarshal.run_validator(
+                            validator, item, original_data[idx], self.fields,
```

Comment:
I'm concerned that this will break for collections that don't support indexing (e.g. QuerySets in Django). I think @lafrech 's suggestion [here](https://github.com/marshmallow-code/marshmallow/issues/743#issuecomment-368141351) might address that.